plugins/cost-insights/src/hooks/useLastCompleteBillingDate.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {
+  PropsWithChildren,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { Alert } from '@material-ui/lab';
+import { useApi } from '@backstage/core';
+import { costInsightsApiRef } from '../api';
+import { MapLoadingToProps, useLoading } from './useLoading';
+import { DefaultLoadingAction, Maybe } from '../types';
+
+type BillingDateProviderLoadingProps = {
+  dispatchLoadingBillingDate: (isLoading: boolean) => void;
+};
+
+const mapLoadingToProps: MapLoadingToProps<BillingDateProviderLoadingProps> = ({
+  dispatch,
+}) => ({
+  dispatchLoadingBillingDate: (isLoading: boolean) =>
+    dispatch({ [DefaultLoadingAction.LastCompleteBillingDate]: isLoading }),
+});
+
+export type BillingDateContextProps = {
+  lastCompleteBillingDate: string; // YYYY-MM-DD
+};
+
+export const BillingDateContext = React.createContext<
+  BillingDateContextProps | undefined
+>(undefined);
+
+export const BillingDateProvider = ({ children }: PropsWithChildren<{}>) => {
+  const client = useApi(costInsightsApiRef);
+  const [error, setError] = useState<Maybe<Error>>(null);
+  const { dispatchLoadingBillingDate } = useLoading(mapLoadingToProps);
+
+  const [lastCompleteBillingDate, setlastCompleteBillingDate] = useState<
+    Maybe<string>
+  >(null);
+
+  useEffect(() => {
+    dispatchLoadingBillingDate(true);
+
+    async function getLastCompleteBillingDate() {
+      try {
+        const d = await client.getLastCompleteBillingDate();
```

Comment:
Since `YYYY-MM-DD` is required formatting for the date string, I think we should add some validation so we can fail early.